core/src/main/java/io/openepcis/epc/eventhash/HashNodeComparator.java

```diff
@@ -111,20 +130,57 @@ private int sortUserExtensions(final ContextNode o1, final ContextNode o2) {
   }
 
   // For nested hashnode values loop over its children and get values.
-  private String findChildren(ArrayList<ContextNode> children) {
+  private String findChildren(final ContextNode node) {
+
+    // Sort the children elements as per standard before building single string for sorting from
+    // children elements
+    this.sortMap = TemplateNodeMap.findSortList(node);
+    final HashNodeComparator comparator = new HashNodeComparator(node, standardFieldSort);
+    if (!node.getChildren().isEmpty()) {
+      node.getChildren().sort(comparator);
+    }
+
     final StringBuilder childrenString = new StringBuilder();
+    final StringBuilder extensionString = new StringBuilder();
+    final ArrayList<ContextNode> children = node.getChildren();
 
     if (children != null) {
       children.forEach(
           child -> {
-            if (child != null && child.getValue() == null) {
-              childrenString.append(findChildren(child.getChildren()));
-            } else if (child != null && child.getValue() != null) {
-              childrenString.append(child.getValue());
+            // Sort only based on dedicated epcis field and ignore the extension values present in
+            // children
+            if (child != null
```

Comment:
check for same condition only once, do not repeat it with every decision to be made:
child != null
&& child.getValue() != null
... 
TemplateNodeMap.isEpcisField(child)